ni/nimble
# Add token representing the height of the table with all rows visible

# Pull Request

## 🤨 Rationale

Resolves #1624

## 👩‍💻 Implementation

The table already knows the combined height of all the rows in the table because it uses this value to size the `table-scroll` container. Therefore, I was able to create a token that gets assigned a value within the scope of the table styling that has a value of the size of the height the table needs to show all rows. The token's name is `tableFitRowsHeight`.

The table also now sets a default max-height on itself to keep a client from accidentally using `tableFitRowsHeight` and trying to render thousands of rows at once.

To use the new token, a client would create their table exactly as before and add the following style:
```scss
nimble-table {
    height: $ni-nimble-table-fit-rows-height;
}
```

## 🧪 Testing

- Manually tested in storybook
- Wrote chromatic tests that verify the height of the table is correct when its height is set to the new token value when:
    - There are 5 records, with & without grouping
    - There are 10 records, with & without grouping
    - There are 50 records, with & without grouping

## ✅ Checklist
